File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/core/actual_settings/DarkModeTest.java
package coolpharaoh.tee.speicher.tea.timer.core.actual_settings;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

class DarkModeTest {

    @Test
    void getTextSystem() {
        final DarkMode darkMode = DarkMode.ENABLED;
        assertThat(darkMode.getText()).isEqualTo("enabled");
    }

    @Test
    void getChoiceSystem() {
        final DarkMode darkMode = DarkMode.ENABLED;
        assertThat(darkMode.getChoice()).isOne();
    }

    @Test
    void darkModeFromTextSystem() {
        final DarkMode darkMode = DarkMode.fromText("enabled");
        assertThat(darkMode).isEqualTo(DarkMode.ENABLED);
    }

    @Test
    void darkModeFromTextNotDefined() {
        final DarkMode darkMode = DarkMode.fromText("not defined");
        assertThat(darkMode).isEqualTo(DarkMode.SYSTEM);
    }

    @Test
    void darkModeFromTextNull() {
        final DarkMode darkMode = DarkMode.fromText(null);
        assertThat(darkMode).isEqualTo(DarkMode.SYSTEM);
    }

    @Test
    void darkModeFromChoiceZero() {
        final DarkMode darkMode = DarkMode.fromChoice(1);
        assertThat(darkMode).isEqualTo(DarkMode.ENABLED);
    }

    @Test
    void darkModeFromChoiceMinusOne() {
        final DarkMode darkMode = DarkMode.fromChoice(-1);
        assertThat(darkMode).isEqualTo(DarkMode.SYSTEM);
    }
}

File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/core/settings/DarkModeTest.kt
package coolpharaoh.tee.speicher.tea.timer.core.settings

import coolpharaoh.tee.speicher.tea.timer.core.settings.DarkMode.Companion.fromChoice
import coolpharaoh.tee.speicher.tea.timer.core.settings.DarkMode.Companion.fromText
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class DarkModeTest {
    @Test
    fun textSystem() {
        val darkMode = DarkMode.ENABLED
        assertThat(darkMode.text).isEqualTo("enabled")
    }

    @Test
    fun choiceSystem() {
        val darkMode = DarkMode.ENABLED
        assertThat(darkMode.choice).isOne
    }

    @Test
    fun darkModeFromTextSystem() {
        val darkMode = fromText("enabled")
        assertThat(darkMode).isEqualTo(DarkMode.ENABLED)
    }

    @Test
    fun darkModeFromTextNotDefined() {
        val darkMode = fromText("not defined")
        assertThat(darkMode).isEqualTo(DarkMode.SYSTEM)
    }

    @Test
    fun darkModeFromTextNull() {
        val darkMode = fromText(null)
        assertThat(darkMode).isEqualTo(DarkMode.SYSTEM)
    }

    @Test
    fun darkModeFromChoiceZero() {
        val darkMode = fromChoice(1)
        assertThat(darkMode).isEqualTo(DarkMode.ENABLED)
    }

    @Test
    fun darkModeFromChoiceMinusOne() {
        val darkMode = fromChoice(-1)
        assertThat(darkMode).isEqualTo(DarkMode.SYSTEM)
    }
}